samples/api/hello_triangle_1_3/hello_triangle_1_3.cpp

```diff
@@ -0,0 +1,1290 @@
+	required_instance_extensions.push_back(VK_KHR_XLIB_SURFACE_EXTENSION_NAME);
+#elif defined(VK_USE_PLATFORM_WAYLAND_KHR)
+	required_instance_extensions.push_back(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
+#elif defined(VK_USE_PLATFORM_DISPLAY_KHR)
+	required_instance_extensions.push_back(VK_KHR_DISPLAY_EXTENSION_NAME);
+#else
+#	pragma error Platform not supported
+#endif
+
+	if (!validate_extensions(required_instance_extensions, available_instance_extensions))
+	{
+		throw std::runtime_error("Required instance extensions are missing.");
+	}
+
+	uint32_t instance_layer_count;
+	VK_CHECK(vkEnumerateInstanceLayerProperties(&instance_layer_count, nullptr));
+
+	std::vector<VkLayerProperties> supported_validation_layers(instance_layer_count);
+	VK_CHECK(vkEnumerateInstanceLayerProperties(&instance_layer_count, supported_validation_layers.data()));
+
+	std::vector<const char *> requested_validation_layers{};
+
+#ifdef defined(VKB_DEBUG) || VKB_VALIDATION_LAYERS
+	// Determine the optimal validation layers to enable that are necessary for useful debugging
+	std::vector<const char *> optimal_validation_layers = vkb::get_optimal_validation_layers(supported_validation_layers);
+	requested_validation_layers.insert(requested_validation_layers.end(), optimal_validation_layers.begin(), optimal_validation_layers.end());
+#endif
+
+	if (validate_layers(requested_validation_layers, supported_validation_layers))
+	{
+		LOGI("Enabled Validation Layers:");
+		for (const auto &layer : requested_validation_layers)
+		{
+			LOGI("	\t{}", layer);
+		}
+	}
+	else
+	{
+		throw std::runtime_error("Required validation layers are missing.");
+	}
+
+	VkApplicationInfo app{
+	    .sType            = VK_STRUCTURE_TYPE_APPLICATION_INFO,
+	    .pApplicationName = "Hello Triangle V1.3",
+	    .pEngineName      = "Vulkan Samples",
+	    .apiVersion       = VK_MAKE_VERSION(1, 3, 0)};
+
+	VkInstanceCreateInfo instance_info{
+	    .sType                   = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
+	    .pApplicationInfo        = &app,
+	    .enabledLayerCount       = vkb::to_u32(requested_validation_layers.size()),
+	    .ppEnabledLayerNames     = requested_validation_layers.data(),
+	    .enabledExtensionCount   = vkb::to_u32(required_instance_extensions.size()),
+	    .ppEnabledExtensionNames = required_instance_extensions.data()};
+
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+	VkDebugUtilsMessengerCreateInfoEXT debug_messenger_create_info = {VK_STRUCTURE_TYPE_DEBUG_UTILS_MESSENGER_CREATE_INFO_EXT};
+	if (has_debug_utils)
+	{
+		debug_messenger_create_info.messageSeverity = VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT | VK_DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT;
+		debug_messenger_create_info.messageType     = VK_DEBUG_UTILS_MESSAGE_TYPE_GENERAL_BIT_EXT | VK_DEBUG_UTILS_MESSAGE_TYPE_VALIDATION_BIT_EXT;
+		debug_messenger_create_info.pfnUserCallback = debug_callback;
+
+		instance_info.pNext = &debug_messenger_create_info;
+	}
+#endif
```

Comment:
Hmm, my thought process was i am going to check for Vulkan 1.3 which is not supported on IOS anyway but you are correct i should do portability handling so when V1.3 is available on iOS it should work.

Reply:
Since this sample is specifically targeting Vulkan 1.3-compatible hardware, and the code already checks for Vulkan 1.3 support on the selected device, there's no need to enable the portability extension (VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR). The portability subset is mainly used to ensure Vulkan works on devices with partial or non-standard Vulkan implementations, such as older or embedded systems.

However, in this case, I am focusing on platforms where Vulkan 1.3 is fully supported (modern desktops, mobile devices, etc.). Therefore, enabling the portability flag is unnecessary.

@SaschaWillems  what is your opinion about this ?

Reply:
Agree 100%. MoltenVK can't do 1.3 and it's probably going to take a while before it gets 1.3 support. And I don't think we should delay this sample because of that.

So why not remove the Mac related stuff for now and add support in a different PR once MoltenVK get's support for 1.3.

Reply:
Ok, agreed.